test/DotNetty.Transport.Tests.Performance/Sockets/TcpServerSocketChannelHorizontalScaleStressSpec.cs

```diff
@@ -0,0 +1,196 @@
+        ///     If it takes longer than <see cref="SaturationThreshold" /> to establish 10 connections, we're saturated.
+        ///     End the stress test.
+        /// </summary>
+        static readonly TimeSpan SaturationThreshold = TimeSpan.FromSeconds(15);
+
+        protected readonly ManualResetEventSlim ResetEvent = new ManualResetEventSlim(false);
+        ConcurrentBag<IChannel> clientChannels;
+        Counter clientConnectedCounter;
+        Counter errorCounter;
+
+        Action eventLoop;
+        Counter inboundThroughputCounter;
+        Counter outboundThroughputCounter;
+
+        IChannel serverChannel;
+        CancellationTokenSource shutdownBenchmark;
+
+        IReadFinishedSignal signal;
+
+        protected Bootstrap ClientBootstrap;
+
+        protected IEventLoopGroup ClientGroup;
+        protected ServerBootstrap ServerBoostrap;
+        protected IEventLoopGroup ServerGroup;
+        protected IEventLoopGroup WorkerGroup;
+
+        protected virtual IChannelHandler GetEncoder() => new LengthFieldPrepender(4, false);
+
+        protected virtual IChannelHandler GetDecoder() => new LengthFieldBasedFrameDecoder(int.MaxValue, 0, 4, 0, 4);
+
+        static readonly ThreadLocal<Random> ThreadLocalRandom = new ThreadLocal<Random>(() => new Random());
+
+        [PerfSetup]
+        public void SetUp(BenchmarkContext context)
+        {
+            this.ClientGroup = new MultithreadEventLoopGroup(Environment.ProcessorCount / 2);
+            this.ServerGroup = new MultithreadEventLoopGroup(1);
+            this.WorkerGroup = new MultithreadEventLoopGroup(Environment.ProcessorCount / 2);
+
+            this.shutdownBenchmark = new CancellationTokenSource();
+            this.clientChannels = new ConcurrentBag<IChannel>();
+
+            this.inboundThroughputCounter = context.GetCounter(InboundThroughputCounterName);
+            this.outboundThroughputCounter = context.GetCounter(OutboundThroughputCounterName);
+            this.clientConnectedCounter = context.GetCounter(ClientConnectCounterName);
+            this.errorCounter = context.GetCounter(ErrorCounterName);
+
+            this.signal = new ManualResetEventSlimReadFinishedSignal(this.ResetEvent);
+
+            ServerBootstrap sb = new ServerBootstrap().Group(this.ServerGroup, this.WorkerGroup).Channel<TcpServerSocketChannel>()
+                .ChildOption(ChannelOption.TcpNodelay, true)
+                .ChildHandler(new ActionChannelInitializer<TcpSocketChannel>(channel =>
+                {
+                    channel.Pipeline.AddLast(this.GetEncoder())
+                        .AddLast(this.GetDecoder())
+                        .AddLast(new IntCodec(true))
+                        .AddLast(new CounterHandlerInbound(this.inboundThroughputCounter))
+                        .AddLast(new CounterHandlerOutbound(this.outboundThroughputCounter))
+                        .AddLast(new ErrorCounterHandler(this.errorCounter));
+                }));
```

Comment:
So modify one of these existing handlers to accept the read and end the propagation? Or write a new one that acts as a "blackhole" of sorts?